src/charmed_kubeflow_chisme/bundle/_bundle.py

```diff
@@ -0,0 +1,91 @@
+# Copyright 2022 Canonical Ltd.
+# See LICENSE file for licensing details.
+"""Helper to work with Juju bundles, preserving comments in the YAML"""
+
+import copy
+from pathlib import Path
+from typing import Optional
+
+from deepdiff import DeepDiff
+from ruamel.yaml import YAML
+
+from ..juju import Juju
+
+
+class Bundle:
+    """Juju bundle loader/dumper.
+
+    This class uses ruamel.yaml instead of the typical pyyaml in order to preserve all comments,
+    the order of the yaml, etc.  This way we can load/dump a yaml with comments without losing them
+    or reordering the items.
+    """
+
+    def __init__(self, filename: Optional[str] = None):
+        self._filename = filename
+        self._data = None
+        if self._filename:
+            self.load_bundle()
+
+    def deepcopy(self):
+        """Returns a new deep copy of this Bundle"""
+        newbundle = copy.deepcopy(self)
+        return newbundle
+
+    def diff(self, other):
```

Comment:
```suggestion
    def diff(self, other: Bundle):
```

I am not sure what our type hints policy is. I see some effort here and also in other parts. This is not critical comment.

Reply:
Also not sure about return values for type hints

Reply:
Tried to add annotations everywhere they were missing.  I didn't know till today how to annotate a method on a class with the class itself.  Turns out you need `annotations` from `__futures__`!

Reply:
[Here](https://peps.python.org/pep-0673/) some reading if you will have some long nights :D IMHO I just hate those annotations Guido is making some weird stuff from Python

Reply:
hahaha yeah I like them really but they're a bit complicated... does this look good now?